LF-4481:  Add Location to animal inventory UI

SayakaOno opened the pull request:
**Description**

Inventory table
- Move the Location type to a common file
- Add `location_id` to Animal and AnimalBatch types and adjust mockData
- Update `useAnimalInventory` to include location
- Add Location column to animal inventory table
- Update `useFilteredInventory` to handle locations filter
- Add `location` string to animal inventory searchable string
- Update `Plain` cell component to take `content`(ReactNode) for greater flexibility
- Update "none" text style in the group column to normal font  (it was previously in italic)
- Fix broken [story](http://localhost:6006/?path=/story/components-tables-cells--fake-table-with-cells)

Header
- Modify `AnimalSingleViewHeader`'s props (separated `locationText` from `animalOrBatch`) and adjust [stories](http://localhost:6006/?path=/docs/components-animals-animalsingleviewheader--docs)
- Send locationText to `AnimalSingleViewHeader`

Jira link: https://lite-farm.atlassian.net/browse/LF-4481

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

antsgar commented:
Looks/works great, it's all coming together!! 🙌

Duncan-Brain commented on packages/webapp/src/components/Table/Cell/CellTypes/Plain.tsx:
```diff
@@ -12,15 +12,16 @@
-const Plain = ({ text }: PlainCellProps) => {
-  return <div className={clsx(styles.text, styles.overflowText)}>{text}</div>;
+const Plain = ({ content }: PlainCellProps) => {
+  return <div className={clsx(styles.text, styles.overflowText)}>{content}</div>;
```
There are also these styles which were meant to only specified for text.

Duncan-Brain commented on packages/webapp/src/containers/Animals/Inventory/index.tsx:
```diff
@@ -145,6 +147,27 @@ function AnimalInventory({
+        label: t('ANIMAL.ANIMAL_LOCATIONS').toLocaleUpperCase(),
+        format: (d: AnimalInventory) => (
+          <Cell
+            kind={CellKind.PLAIN}
```
There is a cell kind of "ICON_TEXT" did it not work?

SayakaOno replied:
I just tried it out. We would need to work around the text (the component requires `ICON_NAME`, and the location needs to be the text `Unknown` for undefined locations) and pass custom styles. I think it's better not to use the `Cell`.

Duncan-Brain replied:
Yup up to you!

I think 

```
d.location ? <Cell kind={ICONTEXT} text iconName  ../> : <Cell kind={PLAIN} text /> 
```

Would also work to keep text styles same across cells but I am good with anything.

SayakaOno replied:
The location text style is unique 😅

Duncan-Brain replied:
Hmm 🤔 .  Yeah, it is unique.... Is it designed to be different than all other cells? It seems strange to me to have one cell have different size, color, and weight of text. 

The whole point of `<Cell />` was to enforce the same text and other styles -- its almost useless otherwise haha.

SayakaOno replied:
I believe the difference is intentional. I don’t have a strong sense of design, but it doesn’t seem strange to me :)

Duncan-Brain reviewed (approved):
Everything works great! 

I might have preferred to keep the `PLAIN` cell kind very text specific, and adapt `ICON_TEXT` instead, or not use standardized Cell at all. `<Cell />` is not necessary if making something single use and custom. But I will leave it up to you!

SayakaOno commented:
Thank you for your input Duncan!

I've reverted the changes to the Plain cell component and updated the location column to avoid using `Cell`.
Thank you in advance for re-reviewing!

SayakaOno commented:
I've just resolved conflicts. Thank you! 🙏
